store/mongo/datastore_mongo.go

```diff
@@ -150,6 +150,31 @@ func (db *MongoStore) InsertDevice(ctx context.Context, dev model.Device) error
 	return errors.Wrap(err, "mongo: failed to store device configuration")
 }
 
+func (db *MongoStore) UpsertExpectedConfiguration(ctx context.Context, dev model.Device) error {
+	if err := dev.Validate(); err != nil {
+		return err
+	}
+	collDevs := db.Database(ctx).Collection(CollDevices)
+
+	fltr := bson.D{{
+		Key:   "_id",
+		Value: dev.ID,
+	}}
+
+	update := bson.M{
+		"$set": bson.D{
+			{
+				Key:   "expected",
+				Value: dev.DesiredAttributes,
+			},
+		},
+	}
+
+	_, err := collDevs.UpdateOne(ctx, fltr, update, mopts.Update().SetUpsert(true))
+
+	return errors.Wrap(err, "mongo: failed to store device configuration")
```

Comment:
I don't know if this can be useful, but we can easily return the updated document as well:
```suggestion
	dev := new(model.Device)
	err := collDevs.FindOneAndUpdate(ctx, fltr, update, mopts.
		FindOneAndUpdate().
		SetUpsert(true).
		SetReturnDocument(mopts.After),
	).Decode(dev)
	return dev, errors.Wrap(err, "mongo: failed to store device configuration")
```

Reply:
lets just skip that one for now.